sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -393,36 +420,268 @@ def tune(
+                    old_attr = getattr(training_args, p_name, None)
+                    if old_attr is not None:
+                        value = f"${{trialParameters.{p_name}}}"
+                    setattr(training_args, p_name, value)
+                    p_value.name = p_name
+                    experiment_params.append(p_value)
+                    trial_params.append(
+                        models.V1beta1TrialParameterSpec(name=p_name, reference=p_name)
+                    )
+                elif p_value is not None:
+                    old_attr = getattr(training_args, p_name, None)
+                    if old_attr is not None:
+                        if isinstance(p_value, dict):
+                            # Update the existing dictionary without nesting
+                            value = copy.deepcopy(p_value)
+                        else:
+                            value = type(old_attr)(p_value)
+                    setattr(training_args, p_name, value)
+
+            lora_config = trainer_parameters.lora_config
+            for p_name, p_value in trainer_parameters.lora_config.__dict__.items():
+                if not hasattr(lora_config, p_name):
+                    logger.warning(
+                        f"Training parameter {p_name} is not supported by the current peft."
+                    )
+                    continue
+                if isinstance(p_value, models.V1beta1ParameterSpec):
+                    old_attr = getattr(lora_config, p_name, None)
+                    if old_attr is not None:
+                        value = f"${{trialParameters.{p_name}}}"
+                    setattr(lora_config, p_name, value)
+                    p_value.name = p_name
+                    experiment_params.append(p_value)
+                    trial_params.append(
+                        models.V1beta1TrialParameterSpec(name=p_name, reference=p_name)
+                    )
+                elif p_value is not None:
+                    old_attr = getattr(lora_config, p_name, None)
+                    if old_attr is not None:
+                        value = type(old_attr)(p_value)
+                    setattr(lora_config, p_name, value)
+
+            init_container_spec = utils.get_container_spec(
+                name=STORAGE_INITIALIZER,
+                base_image=STORAGE_INITIALIZER_IMAGE,
+                args=[
+                    "--model_provider",
+                    mp,
+                    "--model_provider_parameters",
+                    json.dumps(
+                        model_provider_parameters.__dict__, cls=utils.SetEncoder
+                    ),
+                    "--dataset_provider",
+                    dp,
+                    "--dataset_provider_parameters",
+                    json.dumps(dataset_provider_parameters.__dict__),
+                ],
+                volume_mounts=[STORAGE_INITIALIZER_VOLUME_MOUNT],
+            )
```

Comment:
I thought, we discussed before that we are going to re-use helper functions from Training SDK .
Am I missing something ?

Reply:
@andreyvelich Apologies for any confusion, and I may have misunderstood your point. Initially, I attempted to reuse helper functions from the Training SDK, but I later realized it might be easier to maintain if we define our own helper functions in Katib, as the parameters for creating containers and pods differ slightly. Do you think it would be better to reuse functions from the Training SDK?

Reply:
>: 1) use a custom objective function, or 2) use external models and datasets.

@helenxie-bit When user sets external model and dataset, we should use PyTorchJob as Trial template, not Job.

Reply:
Got it! I will update it as soon as possible.